src/Compiler/AST/Visitor/ExprConverter.cpp

```diff
@@ -262,55 +266,111 @@ void ExprConverter::ConvertExprImageAccessArray(ExprPtr& expr, ArrayExpr* arrayE
             {
                 /* Get buffer type denoter from array indices of array access plus identifier */
                 auto bufferTypeDen = bufferDecl->GetTypeDenoter()->GetSub(arrayExpr);
-
-                if (!arrayExpr->arrayIndices.empty())
+                if (auto baseBufferTypeDen = bufferTypeDen->As<BaseTypeDenoter>())
```

Comment:
Same problem here with the `BaseTypeDenoter`: it should be `bufferTypeDen->genericTypeDenoter`, but `bufferTypeDen` has not been casted to its sub type yet.